Modules/@babylonjs/react-native-windows/windows/BabylonReactNative/EngineView.cpp

```diff
@@ -2,173 +2,452 @@
+
+			if (properties.IsRightButtonPressed())
+			{
+				_pressedMouseButtons.insert(Babylon::RIGHT_MOUSE_BUTTON_ID);
+				Babylon::SetMouseButtonState(Babylon::RIGHT_MOUSE_BUTTON_ID, true, x, y);
+			}
+		}
+		else
+		{
+			const auto pointerId = point.PointerId();
+			Babylon::SetTouchButtonState(pointerId, true, x, y);
+		}
+	}
+
+	void EngineView::OnPointerMoved(IInspectable const& /*sender*/, PointerEventArgs const& args)
+	{
+		const auto point = args.CurrentPoint();
+		const auto deviceType = point.PointerDevice().PointerDeviceType();
+		const auto position = point.Position();
+		const uint32_t x = position.X < 0 ? 0 : static_cast<uint32_t>(position.X);
+		const uint32_t y = position.Y < 0 ? 0 : static_cast<uint32_t>(position.Y);
+
+		if (deviceType == PointerDeviceType::Mouse)
+		{
+			Babylon::SetMousePosition(x, y);
+		}
+		else
+		{
+			const auto pointerId = point.PointerId();
+			Babylon::SetTouchPosition(pointerId, x, y);
+		}
+	}
+
+	void EngineView::OnPointerReleased(IInspectable const& /*sender*/, PointerEventArgs const& args)
+	{
+		const auto point = args.CurrentPoint();
+		const auto properties = point.Properties();
+		const auto deviceType = point.PointerDevice().PointerDeviceType();
+		const auto position = point.Position();
+		const uint32_t x = position.X < 0 ? 0 : static_cast<uint32_t>(position.X);
+		const uint32_t y = position.Y < 0 ? 0 : static_cast<uint32_t>(position.Y);
+
+		if (point.PointerDevice().PointerDeviceType() == PointerDeviceType::Mouse)
+		{
+			if (!properties.IsLeftButtonPressed() &&
+				_pressedMouseButtons.find(Babylon::LEFT_MOUSE_BUTTON_ID) != _pressedMouseButtons.end())
+			{
+				_pressedMouseButtons.erase(Babylon::LEFT_MOUSE_BUTTON_ID);
+				Babylon::SetMouseButtonState(Babylon::LEFT_MOUSE_BUTTON_ID, false, x, y);
+			}
+
+			if (!properties.IsMiddleButtonPressed() &&
+				_pressedMouseButtons.find(Babylon::MIDDLE_MOUSE_BUTTON_ID) != _pressedMouseButtons.end())
+			{
+				_pressedMouseButtons.erase(Babylon::MIDDLE_MOUSE_BUTTON_ID);
+				Babylon::SetMouseButtonState(Babylon::MIDDLE_MOUSE_BUTTON_ID, false, x, y);
+			}
+
+			if (!properties.IsRightButtonPressed() &&
+				_pressedMouseButtons.find(Babylon::RIGHT_MOUSE_BUTTON_ID) != _pressedMouseButtons.end())
+			{
+				_pressedMouseButtons.erase(Babylon::RIGHT_MOUSE_BUTTON_ID);
+				Babylon::SetMouseButtonState(Babylon::RIGHT_MOUSE_BUTTON_ID, false, x, y);
+			}
+		}
+		else
+		{
+			const auto pointerId = point.PointerId();
+			Babylon::SetTouchButtonState(pointerId, false, x, y);
+		}
+	}
+
+	void EngineView::OnCompositionScaleChanged(SwapChainPanel const& sender, IInspectable const& /*object*/)
+	{
+		if (_compositionScaleX != sender.CompositionScaleX() || _compositionScaleY != sender.CompositionScaleY())
+		{
+			critical_section::scoped_lock lock(s_criticalSection);
+
+			// Store values so they can be accessed from a background thread.
+			_compositionScaleX = sender.CompositionScaleX();
+			_compositionScaleY = sender.CompositionScaleY();

```

Comment:
When testing this locally, have you ever seen a composition scale != 1? When testing this out on my end I see the correct visual scaling when this composition scale has a value of 1. After initialization this value gets set to 2, which then causes the content to display incorrectly.

Reply:
For what its worth, the swap chain panel size changed event already reports width/height in pixels.

Reply:
it seems like there may be issues here related to: https://github.com/BabylonJS/BabylonNative/pull/582

When i compare this to existing behaviors, this composition scaling change now occurs after babylon.js is initialized. Previously, the composition scale was non one before initialization, which seemed to successfully get relayed to the engine. I'll continue to see whether we can figure out how to fix this.

Reply:
it seems like we may need to call Graphics::SetHardwareScalingLevel to submit this value